test/integration/app-types/src/app/type-checks/link/page.tsx

```diff
@@ -1,9 +1,9 @@
 import type { Route, Metadata } from 'next'
 import Link from 'next/link'
 
-export function Card<T>({ href }: { href: Route<T> | URL }) {
+export function Card<T extends string>({ href }: { href: Route<T> | URL }) {
   return (
-    <Link href={href}>
+    <Link<T> href={href}>
```

Comment:
I've considered this approach but unfortunately we'll have this which isn't ideal. I'm aware of why it's required, but would be great to not have it since href is already a `Route<T>` type, `Link` should just accept that.

Reply:
It just doesn't accept `href={href}` without that, though I don't clearly see why TypeScript doesn't accept that.
Perhaps we should just let `LinkProps` be just like the original then... But I think `Route` should be stricter (I was seeing `any` when it wasn't given a generic).